### ISSUE TITLE
2.6.3 - Pagination bug

### ISSUE THREAD

**Issue by K4T** [cakephp/cakephp#6090]
```php
echo $this->Paginator->sort('id', null, array('model' => 'PdfDocumentPub'));
```

That code will display following errors:

```
Warning (2): array_filter() expects parameter 1 to be array, null given [CORE\Cake\View\Helper\PaginatorHelper.php, line 430]
Warning (2): array_merge() [function.array-merge]: Argument #1 is not an array [CORE\Cake\View\Helper\PaginatorHelper.php, line 430]
```

https://github.com/cakephp/cakephp/blob/master/lib/Cake/View/Helper/PaginatorHelper.php#L429

In that line $paging is null - this is why sort() method is broken when user specify array with model element in last parameter.

Is there a way which allow me to solve this?

**Comment by lorenzo** [comment]
Pass the array as the second argument instead of the third.

**Comment by K4T** [comment]
```php
echo $this->Paginator->sort('id', array('model' => 'PdfDocumentPub'));
```
This will result with: 
```
Notice (8): Array to string conversion [CORE\Cake\View\Helper\HtmlHelper.php, line 373]
```

because second argument must be a string.

**Comment by dereuromark** [comment]
@lorenzo The second argument is the `title`

**Comment by markstory** [comment]
@K4T Did you paginate the PdfDocumentPub model in your controller action?

**Comment by dereuromark** [comment]
We should still catch the notice.
If the model is not available, `$paging = $this->params($model);` returns null. Afterwards it is blindly used as array: `$paging['options']`.

Maybe we can use it conditionally: `!empty($paging['options']) ? $paging['options'] : []` etc

**Comment by markstory** [comment]
Or params() can always return an array. That would be better I think.

**Comment by K4T** [comment]
@markstory: nope, in my controller I am paginating different model:

```php
        $this->paginate = array(
            'PdfPlace' => array(
                'limit' => 26,
                'paramType' => 'querystring'
            )
        );
        $this->set('recentNewspapers', $this->paginate('PdfPlace'));
```
Of course there is association between PdfPlace model and PdfDocumentPub model.

**Comment by markstory** [comment]
Ok, knowing that gives me a place to start looking.

**Comment by markstory** [comment]
Fixed with #6093
